src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -27,7 +27,7 @@
     }
 
     public boolean existsByUsernameOrEmailAndPassword(String usernameOrEmail, String encryptedPassword) {
-        return userRepository.existsByUsernameOrEmailAndPassword(usernameOrEmail, usernameOrEmail, encryptedPassword);
+        return existsByUsername(usernameOrEmail) || existsByEmailAndPassword(usernameOrEmail, encryptedPassword);
```

Comment:
This method returns true if the first condition is true, so it doesn't check in this case password at all, doesn't it?

Reply:
Yes. Parent method wants to know is any of conditions is true.

Reply:
Your method lost primary behavior - so if the user exist it doesn't check it's password. I already told you that you should check @ symbol into the gained usernameOrEmail attribute and invoke existsByUsernameAndPasword or existsByEmailAndPassword

Reply:
@rpkyrych I don't understand why creating two new methods `existsByUsernameAndPasword` and `existsByEmailAndPassword` is better than 
```
findByUsernameOrEmail(usernameOrEmail)
                .filter(u -> isPasswordMatches(u, encryptedPassword))
                .isPresent();
```